pkg/pipelinerunmetrics/metrics_test.go

```diff
@@ -369,14 +373,108 @@ func TestRecordRunningPipelineRunsCount(t *testing.T) {
+	if err := metrics.RunningPipelineRuns(prInformer.Lister(), trInformer.Lister()); err != nil {
 		t.Errorf("RunningPipelineRuns: %v", err)
 	}
 	metricstest.CheckLastValueData(t, "running_pipelineruns_count", map[string]string{}, 1)
 }
 
+func TestRecordRunningPipelineRunsThrottledCounts(t *testing.T) {
+	unregisterMetrics()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	prInformer := fakepipelineruninformer.Get(ctx)
+	trInformer := faketaskruninformer.Get(ctx)
+
+	newTaskRun := func(status corev1.ConditionStatus, reason string) []*v1beta1.TaskRun {
+		retArray := []*v1beta1.TaskRun{}
+		// add 2 taskruns to make sure we don't count a given exceed condition more than once
+		retArray = append(retArray, &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("taskrun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+						Reason: reason,
+					}},
+				},
+			},
+		})
+		retArray = append(retArray, &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("taskrun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+						Reason: reason,
+					}},
+				},
+			},
+		})
+		return retArray
+	}
+	newPipelineRun := func(status corev1.ConditionStatus, reasons []string) *v1beta1.PipelineRun {
+		pr := &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pipelinerun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.PipelineRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+					}},
+				},
+			},
+		}
+		kidRefs := []v1beta1.ChildStatusReference{}
+		for _, reason := range reasons {
+			trs := newTaskRun(status, reason)
+			for _, tr := range trs {
+				// NOTE: while GetIndexer().Add(obj) works well in these unit tests for List(), for
+				// trLister.TaskRuns(trNamespace).Get(trName) to work correctly we need to use GetStore() instead.
+				// see AddToInformer in test/controller.go for an analogous solution
+				if err := trInformer.Informer().GetStore().Add(tr); err != nil {
+					t.Fatalf("Adding TaskRun to trInformer: %v", err)
+				}
+				csr := v1beta1.ChildStatusReference{
+					Name: tr.Name,
+				}
+				csr.Kind = pipeline.TaskRunControllerName
+				kidRefs = append(kidRefs, csr)
+			}
+		}
+		pr.Status.ChildReferences = kidRefs
+		return pr
+	}
+
+	reasons := []string{"", pod.ReasonExceededResourceQuota, pod.ReasonExceededNodeResources}
+
+	// Add N randomly-named PipelineRuns with differently-succeeded statuses.
+	for _, tr := range []*v1beta1.PipelineRun{
```

Comment:
could this use table-driven testing instead?

Reply:
similar rationale to https://github.com/tektoncd/pipeline/pull/6744/files#r1214733673 but sure I'll see about factoring that in as well.